grype/presenter/cyclonedx/presenter.go

```diff
@@ -16,38 +22,49 @@ type Presenter struct {
 	packages         []pkg.Package
 	srcMetadata      *source.Metadata
 	metadataProvider vulnerability.MetadataProvider
+	format           cyclonedx.BOMFileFormat
+	sbom             *sbom.SBOM
 }
 
 // NewPresenter is a *Presenter constructor
-func NewPresenter(results match.Matches, packages []pkg.Package, srcMetadata *source.Metadata, metadataProvider vulnerability.MetadataProvider) *Presenter {
+func NewPresenter(pb models.PresenterBundle, format cyclonedx.BOMFileFormat) *Presenter {
 	return &Presenter{
-		results:          results,
-		packages:         packages,
-		metadataProvider: metadataProvider,
-		srcMetadata:      srcMetadata,
+		results:          pb.Matches,
+		packages:         pb.Packages,
+		metadataProvider: pb.MetadataProvider,
+		srcMetadata:      pb.Context.Source,
+		sbom:             pb.SBOM,
```

Comment:
nit: if we're going to use a `PresenterBundle` should the formats just have a reference to it instead of copying all/most of the properties individually? (same comment for all presenters, which might just reduce a bit of code)

Reply:
I've made a note to refactor this in its own separate PR so it's clear that this is an optimization and not part of the general upgrade to the syft library.